packages/simulation/src/convert.ts

```diff
@@ -0,0 +1,571 @@
+    tempFromState = 0
+    for (const [key, value] of Object.entries(newCombinedStates)) {
+      for (let transElem = 0; transElem < value.length; transElem++) {
+        if (newCombinedStates[key][transElem].symbol in symbolToStatesMap) {
+          for (let toStateElem = 0; toStateElem < newCombinedStates[key][transElem].tostates.length; toStateElem++) {
+            symbolToStatesMap[newCombinedStates[key][transElem].symbol].push([tempFromState, newCombinedStates[key][transElem].tostates[toStateElem]])
+          }
+          tempFromState++
+        } else {
+          // Equal just one state for initialisation, then push the rest
+          symbolToStatesMap[newCombinedStates[key][transElem].symbol] = [[tempFromState, newCombinedStates[key][transElem].tostates[0]]]
+          for (let toStateElem = 1; toStateElem < newCombinedStates[key][transElem].tostates.length; toStateElem++) {
+            symbolToStatesMap[newCombinedStates[key][transElem].symbol].push([tempFromState, newCombinedStates[key][transElem].tostates[toStateElem]])
+          }
+          tempFromState++
+        }
+      }
+    }
+  }
+
+  // Step 3.5, where the merged states will be added to the initialTransitionTable and current transitions will be modified accordingly. Everything below will do just that
+  // by using reference points to the first combined state
+  const lastKey = Object.keys(initialTransitionTable)[Object.keys(initialTransitionTable).length - 1]
+  let lastKeyNumber = parseInt(lastKey)
+  const lastKeyArray = []
+  // Increase by one to get to next key (the new key for initial table)
+  lastKeyNumber += 1
+  // Go through and look through the initial transition table for a specific pattern of transitions that match the state combination.
+  // If found, remove those transitions from the initial transition table and replaces them with a single transition that goes to a new state represented by a new state id
+  for (let mergedElem = 0; mergedElem < Object.keys(newCombinedStates).length; mergedElem++) {
+    const stateArray = Object.keys(newCombinedStates)[mergedElem].split(',').map(Number)
+    for (let curState = 0; curState < numberOfNFAStates; curState++) {
+      if (initialTransitionTable[curState] !== undefined) {
+        for (let curSymbolElem = 0; curSymbolElem < symbolsArray.length; curSymbolElem++) {
+          const filtered = initialTransitionTable[curState].filter(obj => obj[1] === symbolsArray[curSymbolElem])
+          // In this case there is a chance that these transitions are what need to be deleted as they have been merged
+          if (filtered.length > 1 && filtered.length === stateArray.length) {
+            filtered.sort((a, b) => a[0] - b[0])
+            let sameTransitions = true
+            for (let checkElem = 0; checkElem < filtered.length; checkElem++) {
+              if (filtered[checkElem][0] !== stateArray[checkElem]) {
+                sameTransitions = false
+              }
+            }
```

Comment:
```suggestion
            const sameTransitions = !filtered.some((item, i) => item[0] !== stateArray[i])
```
`some` could also be used here like this. My logic might be wrong so double check

Reply:
Would the alternate be true? 
Like remove the double negation and make it  `filtered.some((item, i) => item[0] === stateArray[i])` cause that reads better imo

Reply:
Nah they're not the same, the original code and your first suggestion both ensure that the filtered array and the state array have the same to states (assuming they're sorted), if even one is different then it has to return false because it isn't a state that needs to be merged. The second only checks to see if one pair is the same rather than all of them.

Reply:
Ah I see, so you'd need to use `every` instead of `some`

Alright thanks 👌